### PR TITLE
feat(csv-json-reader): Add a JSON and CSV File loader

### PR DESCRIPTION
**Pull request by Lordnoname**
feat(json/csv-reader): Add JSONReader and begin with CSVReader

feat(csv): Finish CSVLoader

### CODE REVIEW

File: src/main/java/de/thielpaul/vertretungsplan/input/json/JSONSubjectLoader.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+		PATH.mkdirs();
+
+		if (fileForm.exists() == false)
+			fileForm.createNewFile();
+	}
+
+	private void fetchData(final Gson gson) throws IOException {
+		final BufferedReader reader = new BufferedReader(new FileReader(fileForm));
+		final Type formType = new TypeToken<List<Subject>>() {
+		}.getType();
```
Comment (DevVersion) [inline src/main/java/de/thielpaul/vertretungsplan/input/json/JSONSubjectLoader.java]: Wrap in one line

File: src/main/java/de/thielpaul/vertretungsplan/util/PathUtils.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package de.thielpaul.vertretungsplan.util;
+
+public class PathUtils {
+	public static String ROOT_DIRECTORY;
+
+	static {
+		if(isWindows()) {
```
Comment (DevVersion) [inline src/main/java/de/thielpaul/vertretungsplan/util/PathUtils.java]: Space expression from actual bracket.

File: src/main/java/de/thielpaul/vertretungsplan/input/json/JSONTeacherLoader.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+		fetchData(gson);
+	}
+
+	private void createFileIfNotExists() throws IOException {
+		final File PATH = new File(path);
+
+		PATH.mkdirs();
+
+		if (fileForm.exists() == false)
+			fileForm.createNewFile();
```
Comment (DevVersion) [inline src/main/java/de/thielpaul/vertretungsplan/input/json/JSONTeacherLoader.java]: Please wrap that statement in a line with the actual expression

File: src/main/java/de/thielpaul/vertretungsplan/input/json/JSONSubjectLoader.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+		fetchData(gson);
+	}
+
+	private void createFileIfNotExists() throws IOException {
+		final File PATH = new File(path);
+
+		PATH.mkdirs();
+
+		if (fileForm.exists() == false)
+			fileForm.createNewFile();
```
Comment (DevVersion) [inline src/main/java/de/thielpaul/vertretungsplan/input/json/JSONSubjectLoader.java]: Wrap statement in one line with your expression

File: src/main/java/de/thielpaul/vertretungsplan/input/json/JSONRoomReader.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+		PATH.mkdirs();
+
+		if (fileForm.exists() == false)
+			fileForm.createNewFile();
+	}
+
+	private void fetchData(final Gson gson) throws IOException {
+		final BufferedReader reader = new BufferedReader(new FileReader(fileForm));
+		final Type formType = new TypeToken<List<Room>>() {
+		}.getType();
```
Comment (DevVersion) [inline src/main/java/de/thielpaul/vertretungsplan/input/json/JSONRoomReader.java]: Wrap in one line

File: src/main/java/de/thielpaul/vertretungsplan/input/json/JSONRoomReader.java [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.reflect.TypeToken;
+
+import de.thielpaul.vertretungsplan.core.interfaces.Room;
+import de.thielpaul.vertretungsplan.core.objects.School;
+
+public class JSONRoomReader {
+	private final School school;
```
Comment (DevVersion) [inline src/main/java/de/thielpaul/vertretungsplan/input/json/JSONRoomReader.java]: As the coding conventions say .... add a line between the class and their variables

File: src/main/java/de/thielpaul/vertretungsplan/input/json/JSONFormLoader.java [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+		PATH.mkdirs();
+
+		if (fileForm.exists() == false)
+			fileForm.createNewFile();
+	}
+
+	private void fetchData(final Gson gson) throws IOException {
+		final BufferedReader reader = new BufferedReader(new FileReader(fileForm));
+		final Type formType = new TypeToken<List<Form>>() {
+		}.getType();
```
Comment (DevVersion) [inline src/main/java/de/thielpaul/vertretungsplan/input/json/JSONFormLoader.java]: Wrap in one line

File: src/main/java/de/thielpaul/vertretungsplan/input/json/JSONFormLoader.java [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+		fetchData(gson);
+	}
+
+	private void createFileIfNotExists() throws IOException {
+		final File PATH = new File(path);
+
+		PATH.mkdirs();
+
+		if (fileForm.exists() == false)
+			fileForm.createNewFile();
```
Comment (DevVersion) [inline src/main/java/de/thielpaul/vertretungsplan/input/json/JSONFormLoader.java]: Put the actual statement in the same line as the expression is

File: src/main/java/de/thielpaul/vertretungsplan/input/json/JSONFormLevelLoader.java [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+		PATH.mkdirs();
+
+		if (fileForm.exists() == false)
+			fileForm.createNewFile();
+	}
+
+	private void fetchData(final Gson gson) throws IOException {
+		final BufferedReader reader = new BufferedReader(new FileReader(fileForm));
+		final Type formType = new TypeToken<List<FormLevel>>() {
+		}.getType();
```
Comment (DevVersion) [inline src/main/java/de/thielpaul/vertretungsplan/input/json/JSONFormLevelLoader.java]: Wrap in one line

File: src/main/java/de/thielpaul/vertretungsplan/input/json/JSONFormLevelLoader.java [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+		fetchData(gson);
+	}
+
+	private void createFileIfNotExists() throws IOException {
+		final File PATH = new File(path);
+
+		PATH.mkdirs();
+
+		if (fileForm.exists() == false)
+			fileForm.createNewFile();
```
Comment (DevVersion) [inline src/main/java/de/thielpaul/vertretungsplan/input/json/JSONFormLevelLoader.java]: Wrap the statement in the same line as the expression is

**Comment by DevVersion** [comment]
Work in general looks very good. 
But as I commented on different code snippets, we should provide perfect code... that's why you should fix that. 

Summarized -> LGTM

**Comment by Lordnoname** [comment]
Thanks.
Now I fixed the style issues in the given code snippets.

**Pull request closed without merging**